### PR TITLE
Directions to Colleges

### PR DESCRIPTION
**Pull request by AbelKel**
**Feature within this PR**
* Users can tap on the directions button to see a route to a college from google maps
* The simulator did not have google maps installed so it ended up using the web version

**Next PR**
* Translation

https://drive.google.com/file/d/1tw17r-lnT0ix9ZYJ4qaKWY4xABt1VWjG/view?usp=sharing

### CODE REVIEW

File: Capstone/Detail View/DetailsViewController.m [hunk 1]
```diff
@@ -153,6 +164,20 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
     LongDetailsViewController *longDVC = [segue destinationViewController];
     longDVC.college = collegeToPass;
 }
+
+- (CLLocationCoordinate2D)getLocation {
+    self->locationManager = [[CLLocationManager alloc] init];
+    self->locationManager.delegate = self;
```
Comment (skhezri) [inline Capstone/Detail View/DetailsViewController.m]: I suggest moving these two initialization statements into the viewDidLoad function!

File: Capstone/Detail View/DetailsViewController.m [hunk 2]
```diff
@@ -58,6 +58,17 @@ - (void)handleDoubleTap:(UITapGestureRecognizer *)sender {
     [self performSegueWithIdentifier:@"moreDetails" sender:self];
 }
 
+- (IBAction)didTapDirectionsToCollege:(id)sender {
+    CLLocationCoordinate2D coordinate = [self getLocation];
+    CLLocationCoordinate2D start = {coordinate.latitude, coordinate.longitude};
+    CLLocationCoordinate2D destination = {[self.college.lat doubleValue], [self.college.longtuide doubleValue]};
+
+    NSString *googleMapsURLString = [NSString stringWithFormat:@"http://maps.google.com/?saddr=%1.6f,%1.6f&daddr=%1.6f,%1.6f",
+                                     start.latitude, start.longitude, destination.latitude, destination.longitude];
```
Comment (skhezri) [inline Capstone/Detail View/DetailsViewController.m]: is start.latitude and start.longitude different from coordinate.latitude and coordinate.longitude?

File: Capstone/Detail View/DetailsViewController.m [hunk 3]
```diff
@@ -58,6 +58,17 @@ - (void)handleDoubleTap:(UITapGestureRecognizer *)sender {
     [self performSegueWithIdentifier:@"moreDetails" sender:self];
 }
 
+- (IBAction)didTapDirectionsToCollege:(id)sender {
+    CLLocationCoordinate2D coordinate = [self getLocation];
+    CLLocationCoordinate2D start = {coordinate.latitude, coordinate.longitude};
+    CLLocationCoordinate2D destination = {[self.college.lat doubleValue], [self.college.longtuide doubleValue]};
+
```
Comment (skhezri) [inline Capstone/Detail View/DetailsViewController.m]: nit: extra space

**Review by skhezri — COMMENTED** [review]
Wow, what an awesome feature!! Great work and thanks for being consistent with the PR size!!

File: Capstone/Detail View/DetailsViewController.m (hunk 2 above)
Comment (AbelKel) [reply Capstone/Detail View/DetailsViewController.m]: Yes. So, the starting location is the user's location(longitude and latitude) with the end location being the colleges coordinates(longitude and latitude).

File: Capstone/Detail View/DetailsViewController.h [hunk 4]
```diff
@@ -7,7 +7,9 @@
 #import "UIKit/UIKit.h"
 #import "College.h"
 NS_ASSUME_NONNULL_BEGIN
-@interface DetailsViewController : UIViewController
+@interface DetailsViewController : UIViewController<CLLocationManagerDelegate, MKMapViewDelegate> {
```
Comment (angelacaq) [inline Capstone/Detail View/DetailsViewController.h]: are there methods from `CLLocationManagerDelegate` & `MKMapViewDelegate` that we need to implement? i see that we set the delegate of `locationManager` to `self`, which usually means that there's some functionality in `self` that `locationManager` delegates to

where is `MKMapViewDelegate` used?

File: Capstone/Detail View/DetailsViewController.h [hunk 5]
```diff
@@ -7,7 +7,9 @@
...
+    CLLocationManager *locationManager;
```
Comment (angelacaq) [inline Capstone/Detail View/DetailsViewController.h]: let's move this — and the interface — to the .m file, as `locationManager` looks like it should be private. you should be able to conform to the delegate protocols & the variable on `@interface DetailsViewController` in the .m file

**Review by angelacaq — COMMENTED** [review]
image

File: Capstone/Detail View/DetailsViewController.m (hunk 1 above)
Comment (AbelKel) [reply Capstone/Detail View/DetailsViewController.m]: Resolved

File: Capstone/Detail View/DetailsViewController.m (hunk 3 above)
Comment (angelacaq) [reply Capstone/Detail View/DetailsViewController.m]: controversial after all the feedback we've been giving, but i think we're at a point where we can add some nuance around spacing, given all the good work at resolving these nits! 

tl;dr: this seems ok to me

we mainly don't want extra spacing in between } & {:
```
// end of function A
}


- (void)someFunction {
```

and we don't want extra spacing at the top and bottom of braces:
```
- (void)someFunction {

   int index = 0;
}
```
```
- (void)someFunction {
   int index = 0;
   [self doSomethingElse];

}
```
```
if (something) {

   int index = 0;
}
```
```
if (something) {
   int index = 0;
   [self doSomethingElse];

}
```

*however*, sometimes it makes sense to have single newlines in between groupings *within* a method: e.g. having a group for variables, and a group for how they're used; or having a group doing action A, then another group doing action B

File: Capstone/Detail View/DetailsViewController.m (hunk 2 above)
Comment (angelacaq) [reply Capstone/Detail View/DetailsViewController.m]: How about we use rename `coordinate` to `startCoordinate` & use `startCoordinate.longitude` & `startCoordinate.latitude` instead of a new variable?

File: Capstone/Detail View/DetailsViewController.h (hunk 4 above)
Comment (AbelKel) [reply Capstone/Detail View/DetailsViewController.h]: I'm using the delegate method of location manager to request and get the user's current location using the delegate methods startUpdatingLocation and requestWhenInUseAuthorization. I accidentally put the MKMapViewDelegate there by accident and overlooked it. It should have not been there.

File: Capstone/Detail View/DetailsViewController.h (hunk 5 above)
Comment (AbelKel) [reply Capstone/Detail View/DetailsViewController.h]: My bad. I rushed this PR as I didn't make much progress on the Translation branch. 
Resolved!

File: Capstone/Detail View/DetailsViewController.m (hunk 2 above)
Comment (AbelKel) [reply Capstone/Detail View/DetailsViewController.m]: Makes sense...resolved!

File: Capstone/Detail View/DetailsViewController.m (hunk 3 above)
Comment (AbelKel) [reply Capstone/Detail View/DetailsViewController.m]: I see! That makes sense. However, I find it kind of odd that there was no extra space on my IDE, but there's an extra space on line 65 in github. Anyway, resolved!
